### PR TITLE
test: add chunk viability test coverage

### LINKED ISSUE #106: [TEST]: Add unit tests for audioProcessing.ts chunk handling logic
### Description

The `audioProcessing.ts` module handles critical audio chunk splitting and formatting logic, but currently has **zero unit test coverage**. This is a core module that affects transcription quality and reliability.

### Task

1. Create a test file `src/__tests__/audioProcessing.test.ts`
2. Write unit tests covering:
   - Audio chunk size validation
   - Blob to base64 conversion
   - Audio format detection and handling
   - Edge cases: empty audio, oversized chunks, invalid formats
   - Timeout behavior for long audio streams
3. Use the existing test framework (Vitest) configured in the project

### Acceptance Criteria
- Test file exists at `src/__tests__/audioProcessing.test.ts`
- At least 5 meaningful test cases
- All tests pass with `npm test`
- `npm run type-check` passes
- No mocking of external APIs needed (test pure functions only)

### Difficulty
🟢 Beginner — Good first issue

### PR DESCRIPTION
**Pull request by Srishti-Gupta74**
## Description

Added missing unit test coverage for `isChunkViable()` in `audioProcessing.test.ts`.

The new tests cover:

* chunks smaller than the minimum size
* chunks equal to the minimum size
* chunks larger than the minimum size
* empty audio chunks
* custom minimum chunk size handling

Fixes #106

## Type of Change

* [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
* [ ] ✨ New feature (non-breaking change that adds functionality)
* [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] 📝 Documentation update
* [ ] 🎨 UI/UX improvement
* [ ] ♻️ Code refactoring (no functional changes)
* [x] 🧪 Tests

## How Has This Been Tested?

Please describe the tests you ran to verify your changes:

* [x] Built the extension with `npm run build`
* [ ] Loaded the extension in Chrome and tested manually
* [ ] Tested on a live Google Meet call
* [ ] Verified no console errors
* [ ] Tested with ElevenLabs API key
* [ ] Tested with OpenAI API key

## Screenshots (if applicable)

N/A

## Checklist

* [x] I have **starred the repository** ⭐ (helps us grow and show your support!).
* [x] My code follows the project's style guidelines (TypeScript, vanilla CSS, monochrome UI)
* [x] I have performed a self-review of my code
* [ ] I have commented my code where necessary
* [x] My code follows the formatting of this project (run `npm run format`)
* [x] ESLint checks pass locally without any errors (run `npm run lint`)
* [x] TypeScript type checks pass locally (run `npm run type-check`)
* [x] My changes generate no new warnings or errors
* [ ] I have updated the documentation if needed

## Summary by CodeRabbit

* **Tests**
  * Added unit tests for audio chunk validation, ensuring proper handling of various chunk sizes and custom configuration scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/shouri123/Late-Meet/pull/124?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by shouri123 — APPROVED** [review]

**Review by aditidixit1503 — CHANGES_REQUESTED** [review]
<p className="text-gray-500">
  Show your hand to the camera to begin detection.
</p>

<sub>Requested changes in [CodeRabbit Change Stack](https://app.coderabbit.ai/change-stack/shouri123/Late-Meet/pull/124)</sub>

**Review by aditidixit1503 — CHANGES_REQUESTED** [review]
<p className="text-gray-400">
  No word generated yet. Click "Add Letter" to start.
</p>

{isLoading && (
  <p className="text-blue-500">
    Detecting hand gestures...
  </p>
)}



{!cameraEnabled && (
  <p className="text-red-500">
    Please allow camera access to continue.
  </p>
)}

<sub>Requested changes in [CodeRabbit Change Stack](https://app.coderabbit.ai/change-stack/shouri123/Late-Meet/pull/124)</sub>

File: src/audioProcessing.test.ts [hunk 1]
```diff
@@ -165,3 +169,37 @@ test("vadThreshold forwarded in message falls back to 0.012 when absent", () =>
     "missing threshold must default to 0.012",
   );
```
Comment (aditidixit1503) [inline src/audioProcessing.test.ts:170]: Please check them

**Review by aditidixit1503 — CHANGES_REQUESTED** [review]
<sub>Requested changes in [CodeRabbit Change Stack](https://app.coderabbit.ai/change-stack/shouri123/Late-Meet/pull/124)</sub>

**Review by diksh2005pandey-oss — APPROVED** [review]
